Core/include/Acts/EventData/TrackStateProxy.hpp

```diff
@@ -168,18 +206,67 @@ class TrackStateProxy {
     return *this;
   }
 
+  /// @}
+
+  /// @anchor track_state_proxy_props
+  /// @name Track state properties
+  ///
+  /// Properties of the track state represented by @c TrackStateProxy.
+  ///
+  /// Many of these methods come in a @c const and a non-@c const version. The
+  /// non-@c const version is only available if you have an instance of
+  /// @c TrackStateProxy that does not have the @c read_only template parameter set to
+  /// @c true, even if you hold it as an lvalue.
+  ///
+  /// The track states each have an in the track state container. The sequence
+  /// of track states is implemented as a one or two-way linked list, which
+  /// uses indices into the same container.
+  ///
+  /// Each track state has a @c previous index, which points at the track state
+  /// immediately preceding. A track state with a @c previous index of @c
+  /// kInvalid is the first (innermost) track state in a track or track
+  /// candidate. This is also referred to as a *stem* at the track level.
+  ///
+  /// During track finding and fitting, track states are usually appended to the
+  /// sequence, populating the @c previous index of the new track state. Combinatorial
+  /// track finding can produce track states which fork in this way, by having
+  /// more than
+  /// one track state with the same @c previous index.
+  ///
+  /// The track states have static, optional and dynamic properties. Static
+  /// properties are always present, and can always be retrieved. Optional
+  /// components use an extra indirection mechanism that coordinates with the
+  /// backend to allow both not having the component set, or sharing it with
+  /// other track states. An example is a branching trajectory from track
+  /// finding which shares the same predicted parameter vector and associated
+  /// covariance.
+  ///
+  /// Optional components are
+  /// - predicted parameters and covariance
+  /// - filtered parameters and covariance
+  /// - smoothed parameters and covariance
+  /// - jacobian
```

Comment:
off topic and generic question: do we actually need the jacobian at a later point?

Reply:
Only for the smoothing, I believe.

Reply:
for the smoothing we actually just use the covariances and not the jacobian

I wonder if we could make the storage of the jacobian optional in this case?

but this is orthogonal for this PR

Reply:
I think we do use the jacobian in the smoothing:

https://github.com/acts-project/acts/blob/858a48ad3ea8aa4f2e333f81957c937685e687c1/Core/src/TrackFitting/GainMatrixSmoother.cpp#L34-L35

Reply:
ups somehow I missed this ... thanks and sorry for the noise!